src/Server/Resource/DoctrineResource.php

```diff
@@ -294,24 +294,19 @@ public function getMultiKeyDelimiter()
         return $this->multiKeyDelimiter;
     }
 
-    /**
-     * @return $this
-     */
-    public function setHydrator(HydratorInterface $hydrator)
+    public function setHydrator(?HydratorInterface $hydrator): void
     {
         $this->hydrator = $hydrator;
-
-        return $this;
     }
 
     /**
      * @return HydratorInterface
      */
-    public function getHydrator()
+    public function getHydrator(): ?HydratorInterface
     {
         if (! $this->hydrator) {
             // FIXME: find a way to test this line from a created API.  Shouldn't all created API's have a hydrator?
-            $this->hydrator = new Hydrator\DoctrineObject($this->getObjectManager(), $this->getEntityClass());
+            $this->hydrator = new DoctrineObject($this->getObjectManager());
```

Comment:
The 2nd argument passed into this constructor has always been incorrect. It was supposed to be whether to use byValue hydration or not.